xbmc/guilib/GUIInfoTypes.h

```diff
@@ -125,6 +126,27 @@ class CGUIInfoLabel
    */
   static std::string ReplaceAddonStrings(const std::string &label);
 
+  typedef boost::function<std::string(const std::string&)> StringReplacerFunc;
+
+  /*!
+   \brief Replaces instances of $strKeyword[value] with the appropriate resolved string
+   \param strInput text to replace
+   \param strKeyword keyword to look for
+   \param func function that does the actual replacement of each bracketed value found
+   \param strOutput the output string
+   \return whether anything has been replaced.
+   */
+  static bool ReplaceDollarKeywordStrings(const std::string &strInput, const std::string &strKeyword, const StringReplacerFunc &func, std::string &strOutput);
```

Comment:
I know that, but IMO we should move the method `ResolveParameters` and the class `ParamReplacer` to CGUIInfoTypes where we already have the other Replacer functionality. You'll be able revert the changes you made to the method signature of `ReplaceString()` and after that you rename the method `ResolveParameters` to `ReplaceParams` and call it in `CGUIIncludes::ResolveParametersForNode()`.

That works the same way except that we do not change the method signature of `ReplaceString` (less code changes + don't make it public) and we put in the param replacer functionality where we have the other ones.